### PR TITLE
Bug 1185504 - [ServiceWorkers][toolkit] Add performance marks to the …

### PR DESCRIPTION
**Pull request by arcturus**
…library

### CODE REVIEW

**Comment by arcturus** [comment]
This change adds performance marks, just marks (no measures), but also modifies the build script to remove those marks by default (and the debug one), like a dist version.
Also adds a new target `gulp debug` that builds the library leaving all the debug parameters and the performance marks.

@lodr could you take a look?

File: lib/simpleofflinecache.js [hunk 1]
```diff
@@ -40,16 +40,19 @@ function SimpleOfflineCache(cacheName, options, missPolicy) {
...
   // just pass through the response and does nothing.
+  performance.mark('soc_' + request.url);
   if (response) {
     return Promise.resolve(response);
   }
 
   var clone = request.clone();
   var _this = this;
-  debug('Handing fetch event: %s', clone.url);
+  debug('Handing fetch event: %s' + clone.url);
```
Comment (lodr) [inline lib/simpleofflinecache.js]: Ey, remove the `%s`.

File: lib/sww.js [hunk 2]
```diff
@@ -2,6 +2,7 @@
 'use strict';
 
 var debug = 1 ? console.log.bind(console, '[ServiceWorkerWare]') : function(){};
```
Comment (lodr) [inline lib/sww.js]: Can you remove the space before `var`, it makes me feel kind of uncomfortable. xP

File: gulpfile.js [hunk 3]
```diff
@@ -21,7 +22,29 @@ gulp.task('tests', function () {
...
+    standAlone: 'ServiceWorkerWare'
+  });
+
+  var bundle = function() {
+    return bundler
+      .bundle()
+      .pipe(source(getBundleName() + '.js'))
+      .pipe(buffer())
+      .pipe(removeLines({'filters': [
+        /performance/,
```
Comment (lodr) [inline gulpfile.js]: Please, specify `.mark()` and `.measure()` explicitely to avoid conflicts with identifiers.

File: lib/sww.js (hunk 2 above)
Comment (arcturus) [inline lib/sww.js]: That's pretty weird, cannot see any space on my editor.

Comment (lodr) [inline lib/sww.js]: Ok. No prob.

**Pull request merged**